docs/help/contributor/style-guide/documenting-code.md

```diff
@@ -0,0 +1,212 @@
+# Documenting Code
+
+??? tip "Looking for a template?"
+    //todo
+
+## Documenting Commands
+
+>**The standard line is “X the Y by running:”**
+
+>It meets these criteria:
+
+>* Explicitly mentions running the command (this isn’t always obvious)
+* **Uses “run”** (and not “type”, “execute”, etc -- we want consistency)
+* Starts with the key information that describes the command, e.g. “To do X...enter Y command:”, “Do X by entering Y command:”
+* As short as possible
+
+> If you must deviate from the standard line, ensure you still meet the above criteria.
+
+=== ":white_check_mark: Correct"
+    Create the service by running:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+=== ":no_entry: Incorrect"
+    **Example 1:**
+
+    Create the service:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 2:**
+
+    Run:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 3:**
+
+    Run the command below to create a service:
+
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+
+## Documenting YAML
+>Use **kubectl apply** for files/objects that the user creates — it works for both “create” and “update”, and the source of truth is their local files.
+Use **kubectl edit** for files which are shipped as part of the Knative software, like the serving/eventing ConfigMaps.
+
+=== ":white_check_mark: Correct"
+
+    Creating a new file:
+    ```
+    kubectl apply -f - <<EOF
+    # code
+    EOF
+    ```
+
+    Editing a file:
+    ```
+    kubectl -n <namespace> edit cm <filename>
+    ```
+
+=== ":no_entry: Incorrect"
+    ```
+    cat <<EOF | kubectl create -f -
+    # code
+    EOF
+    ```
+
+
+## Words requiring code formatting
+Use code formatting to indicate special purpose text. Apply code formatting to the following content:
+
+* Text that the user enters into a field
+* Filenames and path names
+* Configuration file content
+* Sections of code
+* Programming examples
+* Command line text
+* Environment variable text
+
+
+## Referencing Variables in Code Blocks
+
+>Format variables in code blocks like so: <service-name>
+
+> - All lower case
+- Hyphens between words
+- Explanation for each variable below code block
+- Explanation format is “Where... `<service-name>` is…"
+- If there are multiple variables, see below.
```

Comment:
Just a question: any idea when we get callouts properly for mkdocs?

Reply:
how do you mean?

Reply:
Never mind, let's not derail the PR on my internal monologue 😄